lib/modules/datasource/deno/index.ts

```diff
@@ -0,0 +1,117 @@
+import is from '@sindresorhus/is';
+import pMap from 'p-map';
+import { logger } from '../../../logger';
+import { cache } from '../../../util/cache/package/decorator';
+import { regEx } from '../../../util/regex';
+import { joinUrlParts } from '../../../util/url';
+import * as semanticVersioning from '../../versioning/semver';
+import { Datasource } from '../datasource';
+import type { Release } from '../index';
+import type { GetReleasesConfig, ReleaseResult } from '../types';
+import type {
+  DenoAPIModuleResponse,
+  DenoAPIModuleVersionResponse,
+} from './types';
+import { createSourceURL, tagsToRecord } from './utils';
+
+export class DenoDatasource extends Datasource {
+  static readonly id = 'deno';
+
+  override readonly customRegistrySupport = true;
+
+  override readonly registryStrategy = 'first';
+
+  override readonly defaultVersioning = semanticVersioning.id;
+
+  override readonly defaultRegistryUrls = ['https://apiland.deno.dev'];
+
+  constructor() {
+    super(DenoDatasource.id);
+  }
+
+  @cache({
+    namespace: `datasource-${DenoDatasource.id}`,
+    key: ({ packageName, registryUrl }: GetReleasesConfig) =>
+      // TODO: types (#7154)
+      // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+      `${registryUrl}:${packageName}`,
+  })
+  async getReleases({
+    packageName,
+    registryUrl,
+  }: GetReleasesConfig): Promise<ReleaseResult | null> {
+    const massagedRegistryUrl = registryUrl ?? this.defaultRegistryUrls[0];
+
+    const extractResult = regEx(
+      '^(https:\\/\\/deno.land\\/)(?<rawPackageName>[^@\\s]+)'
+    ).exec(packageName);
+    if (is.nullOrUndefined(extractResult)) {
+      logger.debug(
+        `Could not match package regex against packageName: "${packageName}"`
+      );
+      return null;
+    }
+
+    const rawPackageName = extractResult.groups?.rawPackageName;
+    // istanbul ignore if
+    if (is.nullOrUndefined(rawPackageName)) {
+      logger.debug(
+        `Could not extract rawPackageName from package regex: "${packageName}"`
+      );
+      return null;
+    }
+
+    // remove third-party prefix if defined. The only internal library is `std` and is available under the same API
+    const massagedPackageName = rawPackageName.replace('x/', '');
```

Comment:
this will replace all `x/` in name. could this cause wrong side effects?

Reply:
From the jsdoc: 
> If pattern is a string, only the first occurrence will be replaced.

Reply:
yes, but what if that string isn't at packagename start? can that happen?

Reply:
Not for `deno.land` which we check in the regex above.